service/lib/agama/storage/volume_outline.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "pathname"
+require "y2storage/disk_size"
+
+module Agama
+  module Storage
+    # Set of rules and features used to fully define and validate a given volume
+    class VolumeOutline
+      # Whether the volume is mandatory
+      #
+      # If this is true, the list of volumes used by the storage proposal will always contain
+      # this volume or an equivalent one (ie. one with the same mount_path).
+      #
+      # @return [Boolean]
+      attr_accessor :required
+      alias_method :required?, :required
+
+      # Possible filesystem types for the volume
+      #
+      # @return [Array<Y2Storage::Filesystems::Type>]
+      attr_accessor :filesystems
+
+      # Base value to calculate the min size for the volume (if #auto_size is set to true
+      # for that final volume) or to use as default value (if #auto_size is false)
+      #
+      # @return [Y2Storage::DiskSize]
+      attr_accessor :base_min_size
+
+      # Base value to calculate the max size for the volume (if #auto_size is set to true
+      # for that final volume) or to use as default value (if #auto_size is false)
```

Comment:
I am a bit lost here with comments. Both base_min_size and base_max_size mentions that it will be used as default if auto_size is empty. So what happens when it has different value?

Reply:
I would say the documentation is not true for the false case. If auto is false, base sizes are not used at all.

Reply:
Doc updated.